catwalk/tasks/huggingface.py

```diff
@@ -138,3 +138,33 @@ def hfmc_conversion(
 ) -> InstanceConversion:
     # We're doing this in this stupid way because this makes the conversion function picklable.
     return functools.partial(hfmc_convert, **kwargs)
+
+
+@dataclass
+class HFClassificationInstance:
+    id: Optional[str]
+    premise: str
+    hypothesis: str
+    label: int
```

Comment:
In my mind nli is only a specific kind of classification. So perhaps we could either make this more general so as to support classification of single or multiple sentence inputs. Or more simply we could just name this HFNLIInstance to avoid confusion.

Reply:
I made it work with n-way classification. Though to really make this useful, we need a way to instantiate an encoder-only model with an arbitrary number of heads. Right now, it will just print a big warning and give you wrong results if you mismatch the labels and the model size.